### ISSUE TITLE
Restore extension page after reload

### ISSUE THREAD

**Issue by laggingreflex** [microsoft/vscode#33180]
When I install a new extension and reload the window the extension page is lost. I have to go to extensions and find it again. I often install extensions and look at it's README only after it's loaded. 

Would be nice if the extension page(s) restore after reload.

Also restore the Extensions sidebar with the last query which gets lost as well.

**Comment by joaomoreno** [comment]
@bpasero How to achieve this?

**Comment by bpasero** [comment]
@joaomoreno that is possible by registering an editor input factory that can serialise an editor input to JSON and then from there back again. One example of such a factory is the one for files: https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/files/browser/files.contribution.ts#L113

**Comment by Sakerdot** [comment]
How do you serialize IExtension (needed to create ExtensionsInput I believe)?

**Comment by simonjaeger** [comment]
@bpasero @Sakerdot Asks a valid question. Since IExtension has functions, is serializing it the right way? One could serialize the resource URI and then retrieve the data, but that requires reworking other things. Since the "recommended extensions" already has the data. What do you reckon is the best approach to solve this?

**Comment by simonjaeger** [comment]
@joaomoreno Any thoughts?

**Comment by Ankk98** [comment]
Can we just save the id of the extension which has been installed and when the user will press reload button it will set a task to open up the extension page and when vs code will reload it will first check for that task and if its there, it will simply open the page? And if user press reloads after installing multiple extensions then it will open up the page from which user pressed reload button or it can open all pages of all those extensions which were installed in the last session.
After reloading vs code automatically open up the sidebar with extensions tab open maybe it is storing state somewhere and we can use the same.

I am a beginner and wants to work on the feature if that's OK?

**Comment by AlessandroMilan** [comment]
Hello @joaomoreno 
I am looking to start contributing and would like to start off with this issue.
Is this issue still available?

**Comment by joaomoreno** [comment]
cc @sandy081

**Comment by sandy081** [comment]
@punteek This is still open, so you can contribute 👍

**Comment by bhuvinth** [comment]
Hi guys, Looking for a first opportunity to contribute. 
Can I contribute to this? 
Any pointers to start will help big time. Thanks in advance.

**Comment by sandy081** [comment]
@bhuvinth 

- You have to register an editor input factory so that the input can be retrieved after reload. You have to store extension ID while serialising since you cannot store complete extension. See https://github.com/Microsoft/vscode/blob/393b48d4a5de85b6fa1d2ad0a59588d6715e12ea/src/vs/workbench/contrib/extensions/electron-browser/extensions.contribution.ts#L111

- You have to refactor [ExtensionsInput](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/contrib/extensions/common/extensionsInput.ts#L1) class to accept extension id from the deserialised input and change `reload` method to get the extension from [IExtensionsWorkbenchService](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/contrib/extensions/common/extensions.ts#L81)

**Comment by bhuvinth** [comment]
Zuper.. I just read this. I was sick. I am starting to work on this tonight.

**Comment by albelin** [comment]
Hi, I saw this is task is a good first issue and was wondering where I should get started to work on this. Thanks!

**Comment by mjbvz** [comment]
@sandy081 I thought this would be simple but it turns out it is not. The problem I ran into is that `ExtensionsInput` takes a `IExtension` object which cannot be easily serialized and deserialized.

Assigning back to you and unmarking as good first issue. I still think this is worth fixing if possible

**Comment by VSCodeTriageBot** [comment]
We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future. If you disagree and feel that this issue is crucial: we are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Thanks for your understanding, and happy coding!
